met-web/src/components/engagement/new/view/EngagementLoader.tsx

```diff
@@ -0,0 +1,11 @@
+import { Params, defer } from 'react-router-dom';
+import { getEngagement } from 'services/engagementService';
+import { getEngagementIdBySlug } from 'services/engagementSlugService';
+import { getWidgets } from 'services/widgetService';
+
+export const engagementLoader = async ({ params }: { params: Params<string> }) => {
+    const { slug } = params;
+    const engagement = getEngagementIdBySlug(slug ?? '').then((response) => getEngagement(response.engagement_id));
```

Comment:
Are you meaning to run all these in series? I think without the await keyword, some of these will run asynchronously